src/components/SettingsModal.tsx

```diff
@@ -39,19 +46,50 @@ export const SettingsModal = ({ handleCloseModal, isModalOpen }: ISettingsModal)
     shallow,
   );
   const { setSourceCode } = useIntegrationSourceStore();
+  const [nameValidation, setNameValidation] = useState<
+    'default' | 'warning' | 'success' | 'error' | undefined
+  >('default');
   const previousNamespace = usePrevious(localSettings.namespace);
   const [namespaceValidation, setNamespaceValidation] = useState<
     'default' | 'warning' | 'success' | 'error' | undefined
   >('default');
 
   const { addAlert } = useAlert() || {};
 
+  useEffect(() => {
+    const name = metadata.name ? (metadata.name as string) : '';
+    if (settings.name !== name) {
+      setLocalSettings((state) => ({ ...state, name }));
+    }
+  }, [metadata.name]);
+
+  useEffect(() => {
+    const description = metadata.description ? (metadata.description as string) : '';
+    if (settings.description !== description) {
+      setLocalSettings((state) => ({ ...state, description }));
+    }
+  }, [metadata.description]);
+
   useEffect(() => {
     // update settings with the default namespace fetched from the API
     if (settings.namespace === previousNamespace) return;
     setLocalSettings((state) => ({ ...state, namespace: settings.namespace }));
   }, [settings.namespace]);
 
+  const onChangeDescription = (newDesc: string) => {
+    setLocalSettings((state) => ({ ...state, description: newDesc }));
+  };
+
+  const onChangeName = (newName: string) => {
+    setLocalSettings((state) => ({ ...state, name: newName }));
+
+    if (ValidationService.isNameValidCheck(newName)) {
```

Comment:
[Note for myself] Extract this since is used by changing the name through the routes list